src/megaapi_impl.cpp

```diff
@@ -8930,33 +8914,27 @@ bool MegaApiImpl::isSyncable(const char *path, long long size)
     auto localpath = LocalPath::fromPath(utf8path, *fsAccess);
 
     bool result = false;
-    sdkMutex.lock();
+    SdkMutexGuard g(sdkMutex);
     if (size >= 0)
     {
         if (!is_syncable(size))
         {
-            sdkMutex.unlock();
             return false;
         }
     }
 
-    for (sync_list::iterator it = client->syncs.begin(); it != client->syncs.end(); it++)
-    {
-        Sync *sync = (*it);
+    client->syncs.forEachRunningSync([&](Sync* sync) {
+
         if (sync->localnodebypath(NULL, localpath, &parent) || parent)
         {
             if (sync->localdebris.isContainingPathOf(localpath))
             {
-                break;
+                auto temp = localpath.leafName();
+                auto name = temp.toName(*fsAccess, sync->mFilesystemType);
+                result = is_syncable(sync, name.c_str(), localpath);
```

Comment:
Now that we don't break the loop, but iterate through all the syncs, I think we could fake the `result` with any of the other syncs.

Apart from that, not that `temp`, `name`, `result` used to get set when `!sync->localdebris.isContainingPathOf(localpath)` and now it's the other way :thinking:

Reply:
We don't allow nested syncs so we should only get one match maximum anyway - and even if that wasn't the case, the sync we matched first would have been nondeterministic -  but you're right, I didn't reverse the condition on the test for the debris folder that used to `break`, and should have.  I'll fix that.